Refactor swap action button logic into view model

gemdev111 opened the pull request:
Introduces SwapButtonViewModel to encapsulate swap action button state, title, and behavior. Updates SwapScene and SwapSceneViewModel to use the new view model, simplifying button logic and improving testability. Adds unit tests for SwapButtonViewModel and updates SwapSceneViewModel tests. Removes obsolete SwapTests.swift.

images:
<img width="340" height="720" alt="image" src="https://github.com/user-attachments/assets/6b2f28cf-1482-4485-9c91-67a63a72cd8b" />
closes #1135

gemdev111 committed:
Refactor swap action button logic into view model

Introduces SwapButtonViewModel to encapsulate swap action button state, title, and behavior. Updates SwapScene and SwapSceneViewModel to use the new view model, simplifying button logic and improving testability.
Adds unit tests for SwapButtonViewModel and updates SwapSceneViewModel tests.
Removes obsolete SwapTests.swift.

gemdev111 committed:
minor update

gemdev111 committed:
Refactor retry logic in SwapButtonViewModel

Separated retry logic into canRetryQuotes and canRetrySwap for improved clarity and correctness. Updated buttonAction and type properties to use the new retry checks. Added a unit test to cover edge cases in retry logic.

gemcoder21 commented on Features/Swap/Sources/ViewModels/SwapButtonViewModel.swift:
```diff
@@ -0,0 +1,79 @@
+enum SwapButtonAction: Equatable {
+    case retryQuotes
+    case retrySwap
+    case insufficientBalance(String)
```
```suggestion
    case insufficientBalance(asset: Asset)
```

gemcoder21 replied:
or `symbol: String`

gemdev111 committed:
Merge branch 'main' into 1135-swap-action-simplify

gemdev111 committed:
Update SwapButtonViewModel.swift

gemdev111 committed:
Add PreferencesTestKit to PriceAlertService test setup

Updated PriceAlertService mock to include Preferences and PreferencesTestKit for improved test coverage.
Also fixed SwapButtonViewModelTests to use correct argument for insufficientBalance action.
